### PR TITLE
reward: Delete reward.StakingManager

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

- Delete reward.StakingManager in favor of kaiax/staking
- Pass stakingModule into Downloader to handle the 'staking info sync' feature.
  - Downloader uses GetStakingInfoFromDB and PutStakingInfoToDB
- Kept some code - to be removed after kaiax/valset replaces the Istanbul snapshot facility.

Next PR
- clean up DBManager accessors for StakingInfo and TotalSupply

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [ ] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/cn/backend.go [hunk 1]
```diff
@@ -887,10 +880,6 @@ func (s *CN) Start(srvr p2p.Server) error {
 		s.lesServer.Start(srvr)
 	}
 
-	if !s.chainConfig.IsKaiaForkEnabled(s.blockchain.CurrentBlock().Number()) {
-		reward.StakingManagerSubscribe()
```
Comment (hyunsooda) [inline node/cn/backend.go:891]: Has this subscription been moved or completely deleted here?

File: kaiax/staking/interface.go [hunk 2]
```diff
@@ -32,10 +32,11 @@ type StakingModule interface {
 	// This is the most commonly used getter.
 	GetStakingInfo(num uint64) (*StakingInfo, error)
 
-	// GetStakingInfoFromDB returns the staking info from the database.
+	// Directly access the database.
```
Comment (hyeonLewis) [inline kaiax/staking/interface.go:35]:
```suggestion
	// Directly access the database.
	// Note that db access is only effective before Kaia hardfork.
```

File: node/cn/backend.go (hunk 1 above)
Comment (blukat29) [reply node/cn/backend.go:891]: Completely deleted because StakingManager does not exist anymore.

Comment (blukat29) [reply node/cn/backend.go:891]: Yes I deleted it from staking_manager.go

Comment (blukat29) [reply node/cn/backend.go:891]: Oh wait, I think I should implement a similar logic in kaiax/staking that it does WriteStakingInfo in PostInsertBlock. That was the purpose of handleChainHeadEvent().

Comment (blukat29) [reply node/cn/backend.go:891]: PTAL [7ebc36f](https://github.com/kaiachain/kaia/pull/133/commits/7ebc36f5eb0d08216d759d6121d128d439283459)

**Review by hyeonLewis — APPROVED** [review]

File: node/cn/backend.go (hunk 1 above)
Comment (hyunsooda) [reply node/cn/backend.go:891]: `PostInsertBlock` contributeed to its right place :)

**Review by hyunsooda — APPROVED** [review]

**Comment by blukat29** [comment]
Rebased to latest dev.

**Review by ian0371 — APPROVED** [review]

**Pull request merged**